src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private int amountOfBuy = 0;
+    private int amountOfSupply = 0;
 
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(toFileName, createReport(readFromFile(fromFileName)));
+    }
+
+    private String readFromFile(String fileName) {
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {
+            String data = reader.readLine();
+            while (data != null) {
+                builder.append(data).append(System.lineSeparator());
+                data = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write to the file", e);
+
+        }
+        return builder.toString();
+    }
+
+    private String createReport(String dataFromFile) {
+
+        String[] dataArray = dataFromFile.split(System.lineSeparator());
+        String buyOperation = "buy";
+        String supplyOperation = "supply";
```

Comment:
**Use constants**
If you have strange strings or numbers in the code, it’s better to declare them as constants. The name of the constant should display this object’s purpose.